src/github.com/matrix-org/dendrite/syncserver/storage/output_room_events_table.go

```diff
@@ -244,7 +267,14 @@ func rowsToEvents(rows *sql.Rows) ([]gomatrixserverlib.Event, error) {
 		if err != nil {
 			return nil, err
 		}
-		result = append(result, ev)
+		if order == rowOrderForward {
+			result = append(result, ev)
+		} else if order == rowOrderReverse {
+			result = append([]gomatrixserverlib.Event{ev}, result...)
```

Comment:
I was under the impression that the `append()` built-in did various fun optimisations for prepend operations, but I may be hallucinating this.